### ISSUE TITLE
Flutter tool launches the Flutter Android app with a different intent than Android Studio

### ISSUE THREAD

**Issue by Rickyballs** [flutter/flutter#77869]
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 1.22.2, on Microsoft Windows [Version 6.1.7601], locale zh-CN)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc1)
[√] Android Studio (version 4.0)
[!] Android Studio (version 4.1.0)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
[!] VS Code, 64-bit edition (version 1.18.0)
    X Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[√] Connected device (1 available)

! Doctor found issues in 2 categories.
```

## Steps to Reproduce

https://github.com/Rickyballs/FlutterWebViewBugDemo/tree/launchMode
branch: launchMode

1. Run Application from flutter/native.
2. Click the Home button in the lower right corner of the screen to open the plug.
3. Click home to get app back to the background
4. Click the app icon to bring the app to front.

before:launcherAcitvity(standard)-finish->MainActivity(singleTask)->Plugin/Activity(standard)
**Expected results:** 
native launch:MainActivity(singleTask)->Plugin/Activity(standard)
flutter launch:MainActivity(singleTask)->Plugin/Activity(standard)

**Actual results:** 
native launch:MainActivity(singleTask)->Plugin/Activity(standard)
flutter launch:MainActivity(singleTask)

<details>
  <summary>Logs</summary>
launch from Flutter

```
        Hist #1: ActivityRecord{faf0962 u0 com.ricky.web_app/.MainActivity t1971}
          Intent { flg=0x10000000 cmp=com.ricky.web_app/.MainActivity }
          ProcessRecord{67ba46b 28216:com.ricky.web_app/u0a472}
        Hist #0: ActivityRecord{31ee1e1 u0 com.ricky.web_app/.LaunchActivity t1971}
          Intent { act=android.intent.action.RUN flg=0x30000000 cmp=com.ricky.web_app/.LaunchActivity (has extras) }
          ProcessRecord{67ba46b 28216:com.ricky.web_app/u0a472}
```

launch from Native

```
        Hist #2: ActivityRecord{b09a063 u0 com.ricky.web_app/com.sangcomz.fishbun.ui.album.AlbumActivity t1973}
          Intent { cmp=com.ricky.web_app/com.sangcomz.fishbun.ui.album.AlbumActivity bnds=[79,1364][247,1532] }
          ProcessRecord{dd539b9 28604:com.ricky.web_app/u0a472}
        Hist #1: ActivityRecord{726f91b u0 com.ricky.web_app/.MainActivity t1973}
          Intent { flg=0x10000000 cmp=com.ricky.web_app/.MainActivity }
          ProcessRecord{dd539b9 28604:com.ricky.web_app/u0a472}
        Hist #0: ActivityRecord{b3c2c2 u0 com.ricky.web_app/.LaunchActivity t1973}
          Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 cmp=com.ricky.web_app/.LaunchActivity }
          ProcessRecord{dd539b9 28604:com.ricky.web_app/u0a472}
```

The cat and flg are different.

</details>

I just want to know where this flutter launch setting is configured

**Comment by darshankawar** [comment]
@Rickyballs 
This is doesn't seem to be a bug or a feature request, I suggest you to ask this question on support channels like StackOverflow to get help on it.
Closing from here.

**Comment by xster** [comment]
@Rickyballs your question isn't stated very clearly. You'd have to state a bit more what those terms are. Describe the app a bit. What is native startup and flutter startup. What plugins are they. What are the 2 activities. Then describe a bit what the specific issue is (ideally using concrete code or class names) and describe why it's an issue with your business use case, and why you think it's a bug in the Flutter framework or engine.

**Comment by Rickyballs** [comment]
@xster 
从IDE flutter工程处运行APP-》打开带Activity的插件-》HOME键最小化-》点击桌面图标打开-》重新走了LaunchPage
从IDE android工程处点击运行APP-》打开带Activity的插件-》HOME键最小化-》点击桌面图标打开-》显示了最小化前的页面

**Comment by xster** [comment]
ahhh I understand. You're saying https://github.com/flutter/flutter/blob/e8c3ce6bd1254d90a942c5b5da35b5c160877f00/packages/flutter_tools/lib/src/android/android_device.dart#L618 is a different intent being launched than what Android Studio does when you select to launch it as an Android app instead of a Flutter app? 

-------

你是说作为 Flutter 应用启动的时候，flutter_tools 启动时使用的 intent 在https://github.com/flutter/flutter/blob/e8c3ce6bd1254d90a942c5b5da35b5c160877f00/packages/flutter_tools/lib/src/android/android_device.dart#L618跟通过 Android Studio 启动一个安卓应用用的 intent 不一样是吗？

**Comment by xster** [comment]
What's your ideal solution? For Flutter to use the same intent as Android Studio?

------------

你期待的行为时 Flutter 跟 Android Studio 用同一个 intent 吗？

**Comment by xster** [comment]
The ask is to launch with `Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] }` like Android apps via Android Studio rather than `Intent { act=android.intent.action.RUN }` currently. 

@jonahwilliams what do you think?

**Comment by jonahwilliams** [comment]
I don't really have full context in the original choice here, but as long as it doesn't break anything I don't see the issue

**Comment by Rickyballs** [comment]
@jonahwilliams  If it is not set in this way, people who come into contact with the android launch differences will have to adapt and learn from it。

**Comment by jonahwilliams** [comment]
What is the impact of launching with a different intent?

**Comment by xster** [comment]
> I don't really have full context in the original choice here

I don't either. Though it appears to have been this way from over 5 years ago https://github.com/flutter/flutter/pull/1334. It may have been arbitrary.

**Comment by Rickyballs** [comment]
@jonahwilliams The LauncherActivity will be launched by clicking app icon when enter background.The native way doesn't.

**Comment by gnprice** [comment]
It looks like this was fixed by #104135:
```diff
     final List<String> cmd = <String>[
       'shell', 'am', 'start',
-      '-a', 'android.intent.action.RUN',
+      '-a', 'android.intent.action.MAIN',
+      '-c', 'android.intent.category.LAUNCHER',
```

after the related issue #103234 was filed.
